search2/python/search/save/handler.py

```diff
@@ -212,21 +196,21 @@ def _process_query_chunk(query_args: dict, search_client_args: dict, field_names
 
     return resp["hits"]["total"]["value"]
 
-
-# TODO: get max_slices from opensearch index settings
-def go(
+async def go( # pylint:disable=invalid-name
     input_body: dict,
     search_conf: dict,
+    publisher: Publisher,
     max_query_size: int = 10_000,
     max_slices=1024,
     keep_alive="1d",
 ):
+    """ Main function for running the query and writing the output """
     if not input_body['compress']:
         print("\nWarning: still compressing outputs\n")
```

Comment:
should we be using `logging` here instead?

Reply:
That can be a future enhancement